src/main/java/net/minecraftforge/fluids/FluidUtil.java

```diff
@@ -481,20 +488,29 @@ public static FluidActionResult tryPickUpFluid(@Nonnull ItemStack emptyContainer
 
         BlockState state = worldIn.getBlockState(pos);
         Block block = state.getBlock();
-
+        IFluidHandler targetFluidHandler;
         if (block instanceof IFluidBlock)
         {
-            IFluidHandler targetFluidHandler = getFluidHandler(worldIn, pos, side).orElse(null);
-            if (targetFluidHandler != null)
+            targetFluidHandler = new FluidBlockWrapper((IFluidBlock) block, worldIn, pos);
+        }
+        else if (block instanceof IBucketPickupHandler)
+        {
+            targetFluidHandler = new BucketPickupHandlerWrapper((IBucketPickupHandler) block, worldIn, pos);
+        }
+        else
+        {
+            Optional<IFluidHandler> fluidHandler = getFluidHandler(worldIn, pos, side).resolve();
+            if (!fluidHandler.isPresent())
             {
-                return tryFillContainer(emptyContainer, targetFluidHandler, Integer.MAX_VALUE, playerIn, true);
+                return FluidActionResult.FAILURE;
             }
+            targetFluidHandler = fluidHandler.get();
```

Comment:
I moved this handling from IFluidBlock (as it is not the proper handling for an IFluidBlock) to here in an else branch, but I am unsure if this branch should not just be removed instead as querying a tile for its fluid handler doesn't seem like it has to do with getting a fluid from *in world*.